static/components.interfaces.ts

```diff
@@ -110,17 +110,17 @@ export type OutputState = StateWithTree & {
     editor: number; // EditorId
 };
 
-export type ToolViewState = StateWithTree & {
-    compiler: number; // CompilerId
-    editor: number; // EditorId
-    toolId: string;
-    args: unknown;
-    monacoStdin: boolean;
-};
+export type ToolViewState = StateWithTree &
+    ToolState & {
+        id: number; // CompilerID (TODO: Why is this not part of StateWithTree)
```

Comment:
if the TODO is actionable can we tie it to an issue? or if it's just a "here's a weird thing" let's drop the TODO part?

Reply:
Good idea, will make an issue for this